### PR TITLE
Bug Fix: Unhandled Error in GET Program using id, school, and/or major

### PR DESCRIPTION
**Pull request by DaleCS**
This is a bug fix. This controller is needed for fetching available courses for editing degree plans

## The Bug:
In server/controller/program.controller.js, the GET controller for fetching programs using id, school, and/or major wasn't working due to an unhandled error. 

## The Fix:
I fixed the forEach method that traverses the query to properly traverse the res.query object.

### CODE REVIEW

File: server/controller/program.controller.js [hunk 1]
```diff
@@ -92,22 +92,25 @@ programController.get(
...
+      generateServerErrorCode(
+        res,
+        400,
+        e,
+        ID_OR_ANY_OF_THREE_PARAMETERS_IS_REQUIRED
+      );
     }
 
-    const formattedQuery = {};
+    let formattedQuery = {};
```
Comment (calvinqc) [inline server/controller/program.controller.js]: Does formattedQuery change later?

Comment (DaleCS) [reply server/controller/program.controller.js]: Yes. Properties are added to the object later on inside the following forEach callback function.

Comment (calvinqc) [reply server/controller/program.controller.js]: when formattedQuery change means you assign it to some other object.
`formattedQuery = otherObject;`
 But if you're adding key-value to the Object, formattedQuery remain unchanged

Comment (DaleCS) [reply server/controller/program.controller.js]: Oh. I didn't know that's how const on objects work. Thanks. Changed it back to const.

**Review by calvinqc — APPROVED** [review]
LGTM

**Pull request merged**